python/cuml/cuml/internals/array.py

```diff
@@ -1053,9 +1051,7 @@ def from_input(
             else:
                 return X
 
-        if isinstance(
-            X, (dask_cudf.Series, dask_cudf.DataFrame, dd.Series, dd.DataFrame)
-        ):
+        if hasattr(X, "__dask_graph__") and hasattr(X, "compute"):
```

Comment:
I assume this is the canonical way of sniffing out that something is a dask series/df without having to import dask. Fine for me to use it, how do we make it so others who know nothing about dask know to use this pattern? The obvious way would be to make a helper and hope people discover it, do we want to do that? What other options do we have to socialise this?

Reply:
I agree with the sentiment. RIght now this pattern is used twice in cuml. I think if it were to be repeated a bunch more it'd be worth making a utility (or better yet, rethinking our input ingestion). As it stands I find the inlined version more readable and easy to see what counts as a dask object. My old colleague wrote a blogpost that I'd agree with here: https://matthewrocklin.com/blog/work/2019/06/23/avoid-indirection.

Reply:
Works for me. I'm not a fan of layers on layers either, making it a helper was just the first thing that came to my mind that would solve the problem of "how do we get more people to know that this is the way to do it and that they should use. please don't invent your own way!" - which is maybe a social problem and hence we need a social solution, not a tech solution.